api/rate_limiter.go

```diff
@@ -93,19 +96,35 @@ func parseRateLimiterConfiguration(input string) ([]RateLimiterConfiguration, er
 	return configurations, nil
 }
 
-func initRateLimiters(options *Options) ([]filters.RateLimiterMiddleware, error) {
+func initRateLimiters(ctx context.Context, options *Options) ([]filters.RateLimiterMiddleware, error) {
 	var rateLimiters []filters.RateLimiterMiddleware
 	if !options.APISettings.RateLimitingEnabled {
 		return nil, nil
 	}
+
+	var redisStore limiter.Store
+	var err error
+	if options.RedisClient != nil {
+		redisStore, err = redis.NewStore(options.RedisClient)
+		if err != nil {
+			log.C(ctx).Errorf("failed to initialize redis store: %v", err)
+			return nil, err
+		}
+	}
+
 	configurations, err := parseRateLimiterConfiguration(options.APISettings.RateLimit)
 	if err != nil {
 		return nil, err
 	}
 	for _, configuration := range configurations {
+		store := redisStore
+		if store == nil {
```

Comment:
this log will be logged X number of rate limiters, we can log it once above the loop